File: src/main/java/com/lucas/matches/service/domain/LicenseType.java
package com.lucas.matches.service.domain;

import java.util.HashMap;
import java.util.Map;

public enum LicenseType {
    TOURNAMENT("TOURNAMENT"),
    MATCH("MATCH");

    private final String licenseTypeText;
    private static final Map<String, LicenseType> map = new HashMap<>();
    LicenseType(String licenseTypeText) {
        this.licenseTypeText = licenseTypeText;
    }
    static {
        for (LicenseType licenseType : LicenseType.values()) {
            map.put(licenseType.licenseTypeText, licenseType);
        }
    }

    public static LicenseType value(String licenseTypeText) {
        return map.get(licenseTypeText);
    }
}

File: src/main/kotlin/com/lucas/matches/service/domain/LicenseType.kt
package com.lucas.matches.service.domain

enum class LicenseType(private val licenseTypeText: String) {
    TOURNAMENT("TOURNAMENT"), MATCH("MATCH");

    companion object {
        private val map: MutableMap<String, LicenseType> = HashMap()
        fun value(licenseTypeText: String): LicenseType? {
            return map[licenseTypeText]
        }

        init {
            for (licenseType in values()) {
                map[licenseType.licenseTypeText] =
                    licenseType
            }
        }
    }
}